content/post/2021-06-28-draw-me-a-project/index.md

```diff
@@ -0,0 +1,267 @@
+3.  *"Reproducibility should be as easy as possible."*
+4.  *"Finding things should also be as easy as possible."*
+
+Now some people find all these advantages by structuring their analyses as R packages. Creating an R package to share code and data you use throughout projects is not subject to debate: it's great! Creating your analysis as a package, with dependencies in `DESCRIPTION`, functions in `R/`, analysis in e.g. a vignette, *is* subject to debate.
+
+The advantages are that when doing that you can re-use or refresh your package development skills, and foremost that you can re-use tools made for package development (like devtools and usethis). There's a paper presenting and promoting the approach, where such packages are called research compendia: [Packaging Data Analytical Work Reproducibly Using R (and Friends)](https://www.tandfonline.com/doi/abs/10.1080/00031305.2017.1375986?journalCode=utas20), Ben Marwick, Carl Boettiger & Lincoln Mullen (2018), The American Statistician, 72:1, 80-88, DOI: \<10.1080/00031305.2017.1375986\>
+
+There are specific tools for building and using research compendia:
+
+-   [rrtools](https://github.com/benmarwick/rrtools) by Ben Marwick. *"The goal of rrtools is to provide instructions, templates, and functions for making a basic compendium suitable for writing a reproducible journal article or report with R."*
+-   [holepunch](https://karthik.github.io/holepunch/) by Karthik Ram. *"holepunch will read the contents of your R project on GitHub, create a DESCRIPTION file with all dependencies, write a Dockerfile, add a badge to your README, and build a Docker image. Once these 4 steps are complete, any reader can click the badge and within minutes, be dropped into a free, live, RStudio server. Here they can run your scripts and notebooks and see how everything works."* (🤫 holepunch works [without the compendium structure](https://github.com/karthik/holepunch#alternate-setup-method) as well.)
+
+Now it's good to know not everyone loves the idea of projects as R packages. Miles McBain wrote a blog post ["Project as an R package: An okay idea"](https://www.milesmcbain.com/posts/an-okay-idea/).
+
+I found this quote quite interesting:
+
+> *"My response to advocates of project as a package is: ==You're wasting precious time making the wrong packages.=="*
+
+> *"Instead of shoehorning your work into the package development domain, with all the loss of fidelity that entails, why aren't you packaging tools that create the smooth {devtools}/{usethis} style experience for your own domain?"*
+
+In my talk my own advice is to use whatever structure you, and your team if you have one, prefers, and to choose a structure that can be created automatically. You could be the one creating the package to create projects, as Miles said -- although he also mentioned the [risk of "bitrot"](https://www.milesmcbain.com/posts/an-okay-idea/#where-to-next) for tools maintained and used by few people.
+
+## How to run your project?
```

Comment:
I was suprised to not see "if it is a shiny app, you could be using {golem} which is opniated framework to organize, develop run and deploy shiny apps" :)

Reply:
added a mention earlier in the post, great point!